vizro-core/src/vizro/tables/dash_table.py

```diff
@@ -0,0 +1,12 @@
+"""Module containing the standard implementation of `dash_table.DataTable`."""
+import pandas as pd
+from dash import dash_table
+
+from vizro.models.types import capture
+
+
+@capture("table")
+def dash_data_table(data_frame: pd.DataFrame, **kwargs):
+    """Standard `dash_table.DataTable`."""
+    kwargs.setdefault("columns", [{"name": i, "id": i} for i in data_frame.columns])
+    return dash_table.DataTable(data=data_frame.to_dict("records"), **kwargs)
```

Comment:
@antonymilne  @maxschulz-COL - preferably I would have kept our CSS styling only in the CSS files, but unfortunately that's not possible. The dash datatable comes with inline styling, so it always has precedence over anything I could eventually target via CSS selectors in the assets folder. The only way how to overwrite the native inline styling is to add an` ! important` tag to the CSS properties which is not feasible, as otherwise the users cannot overwrite it with `style_header` and `style_data `anymore. This is currently the only way I think we can ship the dash data table with our default styling. 

Unfortunately, the default styling would disappear as soon as the users use `style_header` and `style_data`, so if they want to keep the styling they would have to copy/paste dict entries from here. @maxschulz-COL shall we add that to the docs?

@maxschulz-COL - can you also double-check with Joe if he's fine with the default styling and functionality of the table right now? Or if we should turn on more properties by default? Compared to our internal version e.g. it doesn't have sorting and other properties natively turned on, but we could do that e.g. via 

```
kwargs.setdefault("sort_action", "native")
kwargs.setdefault("sort_mode", "single")
```

The `kwargs.setdefault("style_as_list_view", True)` removes the vertical grid lines, which makes it look a bit cleaner. It will then also be closer the proposed design.

```suggestion
    kwargs.setdefault("columns", [{"name": i, "id": i} for i in data_frame.columns])
    kwargs.setdefault("style_as_list_view", True)
    kwargs.setdefault("style_data", {"border_bottom": "1px solid var(--border-subtle-alpha-01)", "height": "40px"})
    kwargs.setdefault(
        "style_header",
        {
            "border_bottom": "1px solid var(--state-overlays-selected-hover)",
            "border_top": "1px solid var(--main-container-bg-color)",
            "height": "32px",
        },
    )
    return dash_table.DataTable(data=data_frame.to_dict("records"), **kwargs)
```

Reply:
Would it work if we put this in CSS with `!important` and then require a user who wants to override that to put their own CSS in an assets folder also with `!important`?  Which `!important` would win out? 😬 This would mean that users can't use the `style_header/data` arguments but would keep CSS out of this file. Overall it's probably a worse solution than the above?

We should be able to fiddle with the current solution so that users can modify subkeys without needing to copy and paste though, just by rethinking how to handle `kwargs` a bit.

Reply:
@antonymilne that would work, the users `! important` tag would have precedence as their files get read in last.

So see it very nicely here, table.css is coming from our static assets folder, images.css is from the user assets folder. 
![Screenshot 2023-10-24 at 19 00 31](https://github.com/mckinsey/vizro/assets/90609403/0c050bd2-26a7-4f70-876d-f9db6fa0b9a0)

But yeah, this would mean that users can't use the `style_header/data` arguments to specify the attributes defined here. But maybe that's okay? Personally, I am fine with the solution of users having to specify these attributes in a CSS file with the `! important ` tag then. What do you guys think?

Reply:
Thanks for checking @huong-li-nguyen! My current thinking is that this is actually a worse solution than setting inline styles because:
* `style_*` parameters are well documented on Dash and so would be good if they worked
* I'm not sure whether setting through CSS would even work correctly with [style priorities](https://dash.plotly.com/datatable/style#styles-priority) which I guess may require the `style_*` arguments
* it looks like AG Grid [has similar arguments](https://dash.plotly.com/dash-ag-grid/styling-rows#conditional-row-styles) which do things that can't be achieved that definitely can't be achieved through pure CSS (the alternating row colouring could be with `:nth-child` but I'm pretty sure the whole `styleConditions` thing can't be

Hence the only real drawbacks I think are:
1. it doesn't feel very elegant
2. "Unfortunately, the default styling would disappear as soon as the users use style_header and style_data, so if they want to keep the styling they would have to copy/paste dict entries from here"

The first of these is ok since there doesn't seem to be a better way. The second of these we can fix by settings defaults less destructively:

```
from collections import defaultdict
from collections.abc import Mapping

def _set_defaults_nested(supplied: Mapping, defaults: Mapping) -> dict:
    supplied = defaultdict(dict, supplied)
    for default_key, default_value in defaults.items():
        if isinstance(default_value, Mapping):
            supplied[default_key] = _set_defaults_nested(supplied[default_key], default_value)
        else:
            supplied.setdefault(default_key, default_value)
    return dict(supplied)

# use in table function as something like this
defaults = {
    "columns": [{"name": i, "id": i} for i in data_frame.columns],
    "style_as_list_view": True,
    "style_data": {"border_bottom": "1px solid var(--border-subtle-alpha-01)", "height": "40px"},
    "style_header": {
        "border_bottom": "1px solid var(--state-overlays-selected-hover)",
        "border_top": "1px solid var(--main-container-bg-color)",
        "height": "32px",
    },
}
kwargs = _set_defaults_nested(kwargs, defaults)
```

In theory this should work for arbitrarily deep defaults dictionaries, e.g.
```
defaults = {"a": {"b": {"c": 1, "d": 2}}}
_set_defaults_nested({"a": {"b": {"c": 5}}}, defaults)
# gives {'a': {'b': {'c': 5, 'd': 2}}}
```
... but I haven't tested it very thoroughly so do write some unit tests for it!

Reply:
Ahh, I like this a lot! And if they can't do want to overwrite these defaults in style_XXX, they can still use the important tag and CSS files. @maxschulz-COL - could you test that out? 🙏

Reply:
Tested this with a mix of new parameters and an overwrite of an existing parameter:

```py
style_header={"backgroundColor": "rgb(30, 30, 30)", "fontWeight": "bold", "color": "white","height": "100px"},
```

The below is the result, which keeps the default formatting (borders), changes the overwritten (height) and add some new stylings. I think this is the most desirable result.

<img width="1473" alt="image" src="https://github.com/mckinsey/vizro/assets/83698606/c2f93aab-b960-4a87-86b4-dd12de4f19c4">